### DISCUSSION TITLE
【代码提示】请问纯前端如何继承代码提示

### DISCUSSION THREAD

**Discussion by MilkWangStudio** [opensumi/core#2995] [Q&A] [answered]
我使用的是纯前端接入方式，用来开发Sql编辑工具，请问代码提示怎么处理呢，我希望能输入库名后自动提示出表名。

**Comment by erha19** [comment]
可以看一下这里 https://github.com/opensumi/core/issues/2811

**Reply by MilkWangStudio** [reply]
感谢回复，这个issue我之前看过，不过跟我的需求不是很相似，我希望使用我的元数据来做自动补全，不是单纯的语法片段补全。例如我先从后端拉取整个库的表结构，然后注册到代码提示。输入库名后，自动补全、提示后面的表名。

Monaco-editor有提供扩展方法，我看opensumi源码中有 MonacoSnippetSuggestProvider 像是用来做代码提示的，可是看不懂该怎么用。

**Reply by MilkWangStudio** [reply]
![image](https://github.com/opensumi/core/assets/15866249/97f44fe3-1f03-4fef-a6fb-8d7318d526c7)
类似这种效果。

**Reply by Ricbet** [reply, marked as answer]
@MilkWangStudio 用这段代码试试，用的是 registerCompletionItemProvider API

```typescript
import * as monaco from '@opensumi/monaco-editor-core/esm/vs/editor/editor.api';

monaco.languages.registerCompletionItemProvider('sql', {
  provideCompletionItems: (model, position, context, token) => {
    const suggestions = [];
    // 在这里处理你的后端接口请求
    return {
      suggestions: suggestions,
    };
  },
});
```

**Reply by MilkWangStudio** [reply]
十分感谢，这样确实可以弹出了，我再研究研究怎么替换数据，不同文件做不同提示。

**Comment by AhkunTa** [comment]
> 我使用的是纯前端接入方式，用来开发Sql编辑工具，请问代码提示怎么处理呢，我希望能输入库名后自动提示出表名。

sql编辑场景在纯前端环境下 ，可以使用 sql 语言服务去获取动态语法
整体功能实现较复杂，首先需实现sql的基础语言服务，当获取表列时再去动态请求数据
内部基于这些实现了一套 sql编辑器的场景，后续预计九月初会将此能力迁移至纯前端插件

**Reply by MilkWangStudio** [reply]
嗯嗯，期待到时候的完整功能！ 我先用monaco的api简单实现一下。
